pkg/policy/api/selector.go

```diff
@@ -213,9 +241,10 @@ func (n *EndpointSelector) SyncRequirementsWithLabelSelector() {
 
 // newReservedEndpointSelector returns a selector that matches on all
 // endpoints with the specified reserved label.
-func newReservedEndpointSelector(ID string) EndpointSelector {
+func newReservedEndpointSelector(ID string) *EndpointSelector {
 	reservedLabels := labels.NewLabel(ID, "", labels.LabelSourceReserved)
-	return NewESFromLabels(reservedLabels)
+	reservedES := NewESFromLabels(reservedLabels)
+	return &reservedES
```

Comment:
Is it necessary to have the explicit variable here? Pointer to a temporary does not work?